### PR TITLE
removed alphabets from Bio/Data/CodonTable.py

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request removes alphabets from `Bio/Data/CodonTable.py`.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Data/CodonTable.py [hunk 1]
```diff
@@ -62,8 +57,8 @@ class CodonTable:
     # Not always called from derived classes!
     def __init__(
         self,
-        nucleotide_alphabet=nucleotide_alphabet,
-        protein_alphabet=protein_alphabet,
+        nucleotide_alphabet=None,
```
Comment (peterjc) [inline Bio/Data/CodonTable.py:60]: How would you feel about changing the argument and attribute names from ``nucleotide_alphabet`` and ``protein_alphabet`` (which were alphabet objects) to something like ``nucleotide_letter`` and `protein_letters`` (which are now presumably expected to be strings or similar)? i.e. Make the API change more explicit?

**Comment by peterjc** [comment]
I was surprised that other than a tiny change in ``Bio/Seq.py`` nothing else needed updating, but it looks like ``test_CodonTable.py`` does not create any instances of the class directly.

File: Bio/Data/CodonTable.py (hunk 1 above)
Comment (mdehoon) [reply Bio/Data/CodonTable.py:60]: Using "alphabet" for a string of letters is consistent with the attribute names used in e.g. `Bio.motifs`, `Bio.Align.PairwiseAligner`, and `Bio.Align.substitution_matrices`.

**Comment by mdehoon** [comment]
> Can you add a docstring describing the new argument types - since they are changing that ought to be included.
> 
> (See also comment suggesting changing the argument names since their input type is changing)

Keep in mind what are the highest priorities ... everything takes time.

File: Bio/Data/CodonTable.py (hunk 1 above)
Comment (peterjc) [reply Bio/Data/CodonTable.py:60]: I suppose I should have made this argument more strongly back then. Oh well.

**Comment by peterjc** [comment]
Documenting changes is important - although direct usage of ``CodonTable`` is going to be less common than many other things we are changing.

**Comment by mdehoon** [comment]
> Documenting changes is important - although direct usage of `CodonTable` is going to be less common than many other things we are changing.

This class has not had a meaningful docstring in 20 years, doesn't get instantiated, is poorly designed, and is likely to be removed in the near future. This is not high priority.

**Pull request merged**